nemo/collections/asr/modules/conformer_encoder.py

```diff
@@ -362,25 +379,26 @@ def set_max_audio_length(self, max_audio_length):
         device = next(self.parameters()).device
         self.pos_enc.extend_pe(max_audio_length, device)
 
-        att_mask = torch.ones(1, max_audio_length, max_audio_length, dtype=torch.bool, device=device)
-        if self.chunk_size is None:
-            if self.att_context_size[0] >= 0:
-                att_mask = att_mask.triu(diagonal=-self.att_context_size[0])
-            if self.att_context_size[1] >= 0:
-                att_mask = att_mask.tril(diagonal=self.att_context_size[1])
-        else:
-            chunk_idx = torch.arange(0, max_audio_length, dtype=torch.int, device=att_mask.device)
-            chunk_idx = torch.div(chunk_idx, self.chunk_size, rounding_mode="trunc")
-            diff_chunks = chunk_idx.unsqueeze(1) - chunk_idx.unsqueeze(0)
-            chunked_limited_mask = torch.logical_and(
-                torch.le(diff_chunks, self.left_chunks_num), torch.ge(diff_chunks, 0)
-            )
-            att_mask = torch.logical_and(att_mask, chunked_limited_mask.unsqueeze(0))
+        if self.use_att_mask:
```

Comment:
How about just check the self_attention_model here and skip the part if the it is rel_pos_local_attn to avoid creating a new member?

Reply:
You may set the att_mask to None when not used and then check if it None or not in other places.